fe/fe-core/src/main/java/com/starrocks/transaction/DatabaseTransactionMgr.java

```diff
@@ -1342,19 +1342,28 @@ protected List<List<Comparable>> getPartitionTransInfo(long txnId, long tableId)
     public void removeExpiredTxns(long currentMillis) {
         writeLock();
         try {
+            StringBuilder expiredTxnMsgs = new StringBuilder(1024);
             int numJobsToRemove = getTransactionNum() - Config.label_keep_max_num;
             while (!finalStatusTransactionStateDeque.isEmpty()) {
                 TransactionState transactionState = finalStatusTransactionStateDeque.getFirst();
                 if (transactionState.isExpired(currentMillis) || numJobsToRemove > 0) {
                     finalStatusTransactionStateDeque.pop();
                     clearTransactionState(transactionState);
                     --numJobsToRemove;
-                    LOG.info("transaction [" + transactionState.getTransactionId() +
-                            "] is expired, remove it from transaction manager");
+                    expiredTxnMsgs.append(transactionState.getTransactionId() + ", ");
+                    if (expiredTxnMsgs.length() > 4096) {
```

Comment:
If the length is larger than 4096, where to print the the txnId removed?

Reply:
Each time length reach 4096, we print once. Finally, we print last extra message.